lib/vanagon/engine/hardware.rb

```diff
@@ -0,0 +1,102 @@
+REDIS_PORT = 6379
+LOCK_USER = "vanagon/#{ENV['USER']}"
+
+
+class Vanagon
+  class Engine
+    # Class to use when building on a hardware device (e.g. AIX, Switch, etc)
+    #
+    class Hardware < Base
+
+      # This method is used to obtain a vm to build upon
+      # For the base class we just return the target that was passed in
+      def select_target
+        @build_host
+      end
+
+      # Steps needed to tear down or clean up the system after the build is
+      # complete. In this case, we'll attempt to unlock the hardware
+      def teardown
+        release_lock
+      end
+
+      def initialize(platform)
+        # see if we can aquire a lock
+        Vanagon::Driver.logger.debug "Hardware engine invoked."
+        @platform = platform
+        super
+        @required_attributes << "build_host"
+        @build_host = platform.build_host
+        @target = @build_host
+        aquire_lock
+      end
+
+      # Lock the build_host as specificied in the platform definition. This
+      # implement a rudimentary backoff sequence in order to wait on the host
+      # to unlock. Note: this polls forever.
+      def aquire_lock
+        # Wait if locked
+        if locked?
+          warn "Unable to aquire lock."
+        end
+        sleepcount = 1
+        psleepcount = 1
+        temp = 0
+        while locked?
+          print "#{@build_host} is locked..."
+          puts "waiting #{sleepcount} seconds."
+          sleep sleepcount
+          temp = sleepcount
+          sleepcount += psleepcount
+          psleepcount = temp
+        end
+        lock_contents = {}
+        lock_contents[:user] = LOCK_USER
+        lock_contents[:time] = Time.now.to_s
+        lock_contents[:reason] = "Vanagon build on #{@build_host}"
+        redis = connect_redis
+        redis.set "node_lock_#{@build_host.split('.')[0]}", lock_contents.to_json
+        puts "Hardware locked: #{@build_host}"
+        Vanagon::Driver.logger.info "Hardware locked: #{@build_host}"
+      end
+
+      # Release the lock on build_host
+      def release_lock
+        redis = connect_redis
+        r = redis.get("node_lock_#{@build_host.split('.')[0]}")
+        if r
+          result  = JSON.parse(r)
+          if result['user'] != LOCK_USER
+            Vanagon::Driver.logger.info "Refusing to unlock. Lock on #{@build_host} is owned by #{result['user']}."
+            warn "Refusing to unlock. Lock on #{@build_host} is owned by #{result['user']}."
+          else
+            r = redis.DEL("node_lock_#{@build_host.split('.')[0]}")
+            Vanagon::Driver.logger.info "Removed hardware lock: #{@build_host}."
+            puts "Removed hardware lock: #{@build_host}."
+          end
+        end
+      end
+
+      # Boolean to figure out if a host is locked.
+      #
+      # @return [Bool] whether or not the host is lockd.
+      def locked?
+        redis = connect_redis
+        return true if redis.get("node_lock_#{@build_host.split('.')[0]}")
```

Comment:
you should be able to just return the redis.get directly. If the key doesn't exist, redis.get will return nil, which is falsey. Otherwise it will return the string, which is truthy.